app/controllers/hiring_staff/vacancies/documents_controller.rb

```diff
@@ -26,4 +30,68 @@ def redirect_if_no_supporting_documents
   def redirect_to_next_step_if_save_and_continue
     redirect_to application_details_school_job_path if params[:commit] == 'Save and continue'
   end
+
+  def add_documents(documents_attributes)
+    vacancy ||= school.vacancies.find(session_vacancy_id)
+    documents_attributes.each do |document|
+      vacancy.documents.create(document)
+    end
+    vacancy
+  end
+
+  def process_documents(params)
+    @file_size_limit = 10 # MB
+    @errors = false
+    documents_array = []
+    if params[:documents]&.any?
+      params[:documents].each do |document_params|
+        document_hash = upload_document(document_params)
+        unless @errors
+          documents_array << document_hash
+        end
+      end
+    end
+    documents_array
+  end
+
+  def upload_document(document_params)
+    document_upload = DocumentUpload.new(
+      upload_path: document_params.tempfile.path,
+      name: document_params.original_filename
+    )
+    if document_params.size / 1024.0 / 1024.0 > @file_size_limit
+      file_size_error(document_params.original_filename)
+    end
+    unless @errors
+      document_upload.upload
+      unless document_upload.safe_download
+        virus_error(document_params.original_filename)
+      end
+      create_document_hash(document_params, document_upload)
+    end
+  end
+
+  def file_size_error(filename)
+    @errors = true
```

Comment:
I wonder if there's a risk, in using a separate variable to track whether there are errors, that it will become decoupled from the actual state of `@documents_form.errors` which is partly determined by ActiveModel.
```ruby
unless @errors 
```
could become (something like):
```ruby
unless @documents_form.errors
```
in order to have a single source of truth about whether there are errors. Maybe this is just a pattern I'm unaware of though.

Reply:
I'm not sure, but I think this is because we don't validate whether we have a virus in the form model.
We wouldn't know if there was a virus unless Google tells us so the method is invoked depending on the response from Google.

Reply:
It looks like the error is created on line 84:
```ruby
@documents_form.errors.add(:base, t('jobs.file_virus_error_message', filename: filename))
```
Does that count as validating in the form model, for your purposes?